### DISCUSSION TITLE
Restore Capabilities and Update Testing Methodologies

### DISCUSSION THREAD

**Discussion by vitodintino** [KelvinTegelaar/CIPP#2300] [Ideas]
In light of the most recent update scenario, two things came to mind:

1. How do we test CIPP updates before rolling them out to production?
2. How do we process a restore of the environment if the front end is down?

Thank you for all of the hard work CIPP team!

**Comment by PCCJustin** [comment]
I hate to be that guy, but we rolled out CIPP recently to my organization and I'm getting pushback due to availability issues...

I believe in CIPP and love it myself, but the team doesn't use it because it's "Down a lot".

**Reply by KelvinTegelaar** [reply]
This is the first outage we've had in 2.5 years, so I'm wondering where that comes from :)

**Comment by KelvinTegelaar** [comment]
> In light of the most recent update scenario, two things came to mind:
> 
> 1. How do we test CIPP updates before rolling them out to production?
> 2. How do we process a restore of the environment if the front end is down?
> 
> Thank you for all of the hard work CIPP team!

As you're not a sponsor and self-hosted, you are completely in control of backups, restores, and when you install updates in what sense. I personally recommend running multiple instances, or a test instance you delete after you update. :)  

Sponsor can always hit up the helpdesk for any of these questions, and how we handle that on their side.
